### PR TITLE
Plane: Soaring: Allow climb back to altitude in FBWB and CRUISE modes

### PR DESCRIPTION
**Pull request by samuelctabor**
This improves soaring behaviour by allow FBWB and CRUISE modes to automatically climb back to altitude and for another gliding period in the case that minimum altitude is reached. Previously the behaviour was to rather unhelpfully RTL when reaching minimum altitude.

This makes soaring much more useful especially for FPV pilots.

Tested in SITL (new behaviour covered in autotest) and in flight. This allows some code simplification as the previous mode is now always restored when exiting thermalling/loiter, rather than being dependent on the returned loiter status from AP_Soaring.

### CODE REVIEW

File: ArduPlane/navigation.cpp [hunk 1]
```diff
@@ -339,9 +339,13 @@ void Plane::update_fbwb_speed_height(void)
...
 #if SOARING_ENABLED == ENABLED
-        if (g2.soaring_controller.is_active() && g2.soaring_controller.get_throttle_suppressed()) {
-            // we're in soaring mode with throttle suppressed
-            set_target_altitude_current();;
+        if (g2.soaring_controller.is_active()) {
+            if (g2.soaring_controller.get_throttle_suppressed()) {
+                // we're in soaring mode with throttle suppressed
+                set_target_altitude_current();
+            } else {
+                target_altitude.amsl_cm = 100*plane.g2.soaring_controller.get_alt_cutoff() + 1000 + AP::ahrs().get_home().alt;
```
Comment (peterbarker) [inline ArduPlane/navigation.cpp:347]: Uh - `+1000` ?

Even if there's a good reason - AFAICS that reason should be in a comment...

File: Tools/autotest/arduplane.py [hunk 2]
```diff
@@ -1779,6 +1782,9 @@ def fly_soaring(self):
         alt_ctf = self.get_parameter('SOAR_ALT_CUTOFF')
         self.wait_altitude(alt_ctf-10, alt_ctf, timeout=600, relative=True)
 
+        # Wait
```
Comment (peterbarker) [inline Tools/autotest/arduplane.py:1785]: What are we waiting for?

File: Tools/autotest/arduplane.py [hunk 3]
```diff
@@ -1791,17 +1797,11 @@ def fly_soaring(self):
...
 
-        self.progress("Waiting for RTL")
-        self.wait_mode('RTL')
-
-        alt_rtl = self.get_parameter('ALT_HOLD_RTL')/100
-
-        # Wait for climb to  RTL.
-        self.progress("Waiting for climb to RTL altitude")
-        self.wait_altitude(alt_rtl-5, alt_rtl+5, timeout=60, relative=True)
+        self.progress("Waiting for climn")
```
Comment (peterbarker) [inline Tools/autotest/arduplane.py:1803]: spelting mistake.

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: ArduPlane/navigation.cpp (hunk 1 above)
Comment (magicrub) [reply ArduPlane/navigation.cpp:347]: Aka... we're allergic to magic numbers

Comment (samuelctabor) [reply ArduPlane/navigation.cpp:349]: Added a comment. Let me know if there is a better way of handling this situation, I don't like magic numbers either

File: Tools/autotest/arduplane.py (hunk 2 above)
Comment (samuelctabor) [reply Tools/autotest/arduplane.py]: Expanded on the comment

File: Tools/autotest/arduplane.py (hunk 3 above)
Comment (samuelctabor) [reply Tools/autotest/arduplane.py]: Thanks, fixed!

File: ArduPlane/navigation.cpp (hunk 1 above)
Comment (peterbarker) [reply ArduPlane/navigation.cpp:349]: Well, this is your library, so it's definitely not a blocker  - and the comment makes it crystal clear what's going on.

But yeah, magic numbers are usually bad :-)

**Review by peterbarker — APPROVED** [review]

**Comment by peterbarker** [comment]
Merged, thanks.

**Comment by robustini** [comment]
Unfortunately, the RTL behavior was most useful in case of VLOS, certainly in FPV better this integration, but personally I would have left the other option as well, settable.

**Pull request merged**